### ISSUE TITLE
Add more options to `trusted-click-element`

### ISSUE THREAD

**Issue by D4niloMR** [uBlockOrigin/uBlock-issues#2917]
### Prerequisites

- [X] I verified that this is not a filter list issue. Report any issues with filter lists or broken website functionality in the [uAssets issue tracker](https://github.com/uBlockOrigin/uAssets/issues).
- [X] This is NOT an issue with [YouTube](https://www.github.com/uBlockOrigin/uAssets/issues/19976), [Facebook](https://www.github.com/uBlockOrigin/uAssets/issues/3367) or [Twitch](https://www.github.com/uBlockOrigin/uAssets/issues/5184).
- [X] This is not a support issue or a question. For support, questions, or help, visit [/r/uBlockOrigin](https://www.reddit.com/r/uBlockOrigin/).
- [X] I performed a [cursory search of the issue tracker](https://github.com/uBlockOrigin/uBlock-issues/issues?q=is%3Aissue) to avoid opening a duplicate issue.
- [X] The issue is not present after disabling uBO in the browser.
- [X] I checked the [documentation](https://github.com/gorhill/uBlock/wiki) to understand that the issue I am reporting is not normal behavior.

### I tried to reproduce the issue when...

- [X] uBO is the only extension.
- [X] uBO uses default lists and settings.
- [X] using a new, unmodified browser profile.

### Description

Some details: https://github.com/AdguardTeam/AdguardFilters/commit/cb4900805ad9972c99cee421bacf825906c49ef5

This isn't be the best solution, but it can provide a quick and easy fix, to later properly fix it.

The skippable primevideo ad may be fixable by others filters, but the main goal is the potential of `trusted-click-element`

> Based on this documentacion and in my own usage, trusted-click-element works fine for primevideo skip ad, but after 10 seconds the scriptlet stop, or if click the element 1 time, so it will either skip 1 ad only or skip none because you waited more than 10 seconds to choose a movie/serie.

With the "expected behavior" below, can be used more reliably to click on timer: https://github.com/uBlockOrigin/uAssets/pull/20314

### A specific URL where the issue occurs.

https://www.primevideo.com/region/na/detail/0F7QZ8SNFP7NSGO3UW77XTFI5F/ref=atv_dp_atf_prime_sd_mv_resume_t1ADAAAAAA0wg0?autoplay=1&t=0

### Steps to Reproduce

1. On `https://www.primevideo.com/region/na/storefront/`
2. Choose a recommended movie.
3. Skippable ad on your face.

### Expected behavior

`trusted-click-element` being able to click on "skip" **n** times without stopping.

`, stay, 1` to stay indefinitily until element is clicked.

And something like `, clicks, numberofclickshere` to the `, stay, 1`

Examples: `, stay, 1, clicks, 3` to stay indefinitily until element is clicked three times
`, stay, 1, clicks, 0` stay indefinitily even after clicking in the element. (Forever)

### Actual behavior

Just click "Skip" once and within 10 seconds.

### uBO version

1.52.3rc16

### Browser name and version

Firefox 119

### Operating System and version

Windows 10 19045

**Comment by BlazeFTL** [comment]
Definitely Would Be Great... Clicking On A Element 3-4 Times With A Delay +  Delay Count Starting After Page Loads Completely

**Comment by GunGunGun** [comment]
+1, as this will be one of the better way to solve Youtube's anti-adblock by clicking Skip instead of blocking their code.

This is a few examples of clicking to skip Youtube:

1. https://raw.githubusercontent.com/0x48piraj/fadblock/master/src/safari/js/fadblock.user.js
2. https://greasyfork.org/en/scripts/386925-youtube-ad-cleaner-include-non-skippable-ads-works
3. https://greasyfork.org/en/scripts/459541-youtube%E5%8E%BB%E5%B9%BF%E5%91%8A-youtube-ad-blocker/

**Comment by peace2000** [comment]
I think it would be good idea to have an option like "stay active until the element is actually clicked".

Another reason is that adding a delay doesn't always work properly on different pc's. I have noticed on `consent.youtube.com` that in the last page where consent options are configured and accepted, the accept button might not be clicked. On my faster pc, I had no issues but on my slower, old pc clicking the last button didn't always work. (I had to switch the YT solution to a different one: https://github.com/uBlockOrigin/uAssets/commit/f9ee52c3a14ed555119cf0e8cfefd7a63d868723)

 _Probably it would be ok if the delay would be big enough, but filter author can't really know what delay works universally for all pc's and on different browsers._ Adding a huge delay "just in case" causes the dialog to be visually visible for too long time for every one and it's a bad solution.

On Slack, there was a discussion about `chrono24.com` GDPR dialog. For @ryanbr and @stephenhawk8054 100 ms delay was ok, but for @JobcenterTycoon, 1000 ms was needed. On my end, 100ms works sometimes... to be on the safe side to click the GDPR dialog properly https://github.com/uBlockOrigin/uAssets/commit/02f1d7117f2c9939a7ec69908cf0046cad3038c5 1000ms was added. (`chrono24.com` could luckily workarounded by hiding the dialog via cosmetic filtering, but still, it's not optimal way as clicker shouldn't need to rely on cosmetic hiding.)

**Comment by plazer1** [comment]
> I think that solution would be that the scriptlet activates asap and stays active until the dialog has been clicked but no more than 30 seconds.

What makes you think a 30-second fixed timeout is sensible? OP describes a use-case where he needs the scriptlet running indefinitely, so a fixed delay -- however large -- would not solve their problem. I myself have a use-case where the pop-up opens at arbitrary times, either upon user action or after a random timeout, so I too need `trusted-click-element` without any timeout (and not even 30 minutes would be sufficient).

IMO what makes sense here is to specify the timeout via a parameter like AdGuard does [[1]](https://github.com/AdguardTeam/Scriptlets/blob/487f00d3b71cc9d4da51870cec4387900b0fbe11/src/scriptlets/trusted-click-element.ts#L59) and improve on the AdGuard code by treating a negative value as "timeout disabled; run indefinitely".

**Comment by stephenhawk8054** [comment]
uBO won't run trusted click indefinitely given `MutationObserver` is being used for the whole document. It's an expensive JS function.

**Comment by gorhill** [comment]
You can just append a large number to your list of steps, that last number will be used as the timeout value, example:

    ...##+js(trusted-click-element, '500, #idofelementtoclick, 86400000')

Steps:
- Wait 500 ms
- Click element matching `#idofelementtoclick`
- Timeout after 1 day

When the last step is not a number, a default of `11000` is used.

**Comment by plazer1** [comment]
> uBO won't run trusted click indefinitely given `MutationObserver` is being used for the whole document. It's an expensive JS function.

It's expensive only if there are many mutations being made to the observed DOM. Even so, why should it be any of your business to decide what's "too expensive" for the user? I as the user know best what's too expensive for me, and having to hack together a userscript (which internally uses -- guess what -- a `MutationObserver`) just because you consider `MutationObserver` to be "expensive" is mighty annoying.

> You can just append a large number to your list of steps, that last number will be used as the timeout value, example:

I already figured it out, and used 2147483647 as that's the largest possible value equal to roughly 25 days. It would be nice to have the new syntax & semantics documented; see #4068.

**Comment by stephenhawk8054** [comment]
The features are only added if it benefits majority of users, not just for your needs. Efficiency is always uBO's priority target. If you don't agree, feel free to fork the projects and adjust it to yourself. FOSS projects are done by people scratching their own itches, not demanding others to do it for you.

And we are the ones who will have to maintain and receive the reports and complaints at the end, not you.

**Comment by uBlock-user** [comment]
> It's expensive only if there are many mutations being made to the observed DOM

Nope, running MO **_indefinitely_** always causes performance issues regardless, something you have no clue about. We can't sacrifice performance just because YOU said so.

>  why should it be any of your business to decide what's "too expensive" for the user?

It's gorhill's business to maintain uBO to its existing capacity and status i.e. -- **Finally, an efficient blocker. Easy on CPU and memory.**, not yours to question us. Uninstall uBO and go with ABP or something other blocker if you don't like this fact.
